.editorconfig

```
@@ -0,0 +1,26 @@
+# EditorConfig helps maintain consistent coding styles for developers working on the same project across various editors and IDEs.
+# See https://editorconfig.org.
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.php]
+indent_style = tab
+indent_size = 4
+
+[*.xml]
+indent_style = space
+indent_size = 4
+
+[*.json]
+indent_style = space
+indent_size = 2
+
+[*.yml]
+indent_style = space
+indent_size = 2
```

We can default to 2 spaces and override where needed.

```suggestion
indent_style = space
indent_size = 2

[*.php]
indent_style = tab
indent_size = 4

[*.xml]
indent_size = 4
```